src/instance.ts

```diff
@@ -16,13 +16,101 @@
+import {ServiceObjectConfig, GetConfig} from '@google-cloud/common';
 import {paginator} from '@google-cloud/paginator';
 import {promisifyAll} from '@google-cloud/promisify';
 import * as extend from 'extend';
-import * as is from 'is';
 import snakeCase = require('lodash.snakecase');
 import {Database} from './database';
+import {google as instanceAdmin} from '../proto/spanner_instance_admin';
+import {google as databaseAdmin} from '../proto/spanner_database_admin';
+import {Operation as GaxOperation} from 'google-gax';
+import {
+  SessionPoolOptions,
+  SessionPoolInterface,
+  SessionPool,
+} from './session-pool';
+import {Spanner} from '.';
+import {ServiceError} from 'grpc';
+import {RequestCallback, PagedResponse} from './common';
+import {Stream} from 'stream';
+
+export type IDatabase = databaseAdmin.spanner.admin.database.v1.IDatabase;
+export type IInstance = instanceAdmin.spanner.admin.instance.v1.IInstance;
+export type ILongrunningOperation = instanceAdmin.longrunning.IOperation;
+export type CreateDatabaseResponse = [
+  Database,
+  GaxOperation,
+  ILongrunningOperation
+];
+export type DeleteInstanceResponse = [instanceAdmin.protobuf.IEmpty];
+export type ExistsInstanceResponse = [boolean];
+export type GetInstanceResponse = [Instance, IInstance];
+export type GetMetadataResponse = [IInstance];
+export type GetDatabasesResponse = PagedResponse<
+  Database,
+  databaseAdmin.spanner.admin.database.v1.IListDatabasesResponse
+>;
+export type SetMetadataResponse = [GaxOperation, ILongrunningOperation];
+
+export interface CreateDatabaseOptions
+  extends databaseAdmin.spanner.admin.database.v1.ICreateDatabaseRequest {
+  poolOptions?: SessionPoolOptions;
+  poolCtor?: SessionPool;
+  schema?: string;
+}
+export interface CreateInstanceRequest extends IInstance {
+  nodes: number;
+}
+export interface GetDatabasesRequest
+  extends databaseAdmin.spanner.admin.database.v1.IListDatabasesRequest {
+  autoPaginate?: boolean;
+  maxApiCalls?: number;
+  maxResults?: number;
+}
+export interface CreateInstanceCallback {
+  (
+    err: ServiceError | null,
+    instance: Instance | null,
+    operation: GaxOperation | null,
+    apiResponse: ILongrunningOperation
+  ): void;
+}
+export interface CreateDatabaseCallback {
+  (
+    err: ServiceError | null,
+    database?: Database | null,
+    operation?: GaxOperation | null,
+    apiResponse?: ILongrunningOperation
+  ): void;
+}
+export interface DeleteInstanceCallback {
+  (err: ServiceError | null, apiResponse: instanceAdmin.protobuf.IEmpty): void;
+}
+export interface ExistsInstanceCallback {
+  (err: ServiceError | null, exists: boolean | null): void;
+}
+export type GetDatabasesCallback = RequestCallback<
+  Database,
+  databaseAdmin.spanner.admin.database.v1.IListDatabasesResponse
+>;
+export interface GetInstanceCallback {
+  (
+    err: ServiceError | null,
+    instance?: Instance,
+    apiResponse?: IInstance
+  ): void;
+}
+export interface GetMetadataCallback {
```

Comment:
These names are probably too generic, especially if other classes have similar methods. I think we want to call these `GetInstanceMetadataCalback`, etc.